cookbooks/bcpc-hadoop/recipes/datanode.rb

```diff
@@ -76,40 +80,46 @@
 end
 
 # Setup datanode and nodemanager bits
-if node[:bcpc][:hadoop][:mounts].length <= node[:bcpc][:hadoop][:hdfs][:failed_volumes_tolerated]
-  Chef::Application.fatal!("You have fewer #{node[:bcpc][:hadoop][:disks]} than #{node[:bcpc][:hadoop][:hdfs][:failed_volumes_tolerated]}! See comments of HDFS-4442.")
+ruby_block "verify Hadoop mounts" do
+  block do
+    if node[:bcpc][:storage][:mounts].length <= node[:bcpc][:hadoop][:hdfs][:failed_volumes_tolerated]
```

Comment:
If the machine looses too many disks and drops below `node[:bcpc][:hadoop][:hdfs][:failed_volumes_tolerated]` disks, we will refuse to Chef anything beyond this in the Chef converge phase. I don't know if we want a `raise` or simply a `Chef::Log.warn` as HDFS will do the right thing -- and this was to prevent manual mis-specification.